Add support for reordering notification icons via drag-and-drop

xoascf opened the pull request:
Saves and restores the layout order. It also changes the default behavior: it will click and activate the notification icon immediately after you release the left mouse button.

* Closes #1169
* Closes #1207

xoascf committed:
Draggable notification icons

kuku00GT commented:
Hi, will be this (drag-dropp icon ordering) finished soon and added to official release?

xoascf commented on RetroBar/Controls/NotifyIconList.xaml.cs:
```diff
@@ -197,18 +196,55 @@ private void NotifyIconList_OnUnloaded(object sender, RoutedEventArgs e)
+                .ToList();
+
+            // Sort icons according to saved order
+            var sortedIcons = icons.OrderBy(i => Settings.Instance.NotifyIconOrder?.IndexOf(i.GetInvertIdentifier()) ?? -1).ToList();
```
@copilot That... is expected, this is a taskbar, the new notification icons should appear in the left.

xoascf commented on RetroBar/Controls/NotifyIconList.xaml.cs:
```diff
@@ -197,18 +196,55 @@ private void NotifyIconList_OnUnloaded(object sender, RoutedEventArgs e)
+            {
+                pinnedNotifyIcons.Add(icon);
+            }
+            foreach (var icon in NotificationArea.PinnedIcons.Cast<ManagedShell.WindowsTray.NotifyIcon>().OrderBy(i => Settings.Instance.NotifyIconOrder?.IndexOf(i.GetInvertIdentifier()) ?? -1))
+            {
+                pinnedNotifyIcons.Add(icon);
+            }
```
They should remain there, hopefully.

xoascf commented on RetroBar/Utilities/NotifyIconDropHandler.cs:
```diff
@@ -0,0 +1,48 @@
+{
+    public class NotifyIconDropHandler : IDropTarget
+    {
+        private NotifyIconList _notifyIconList;
```
That shouldn't happen...

xoascf commented on RetroBar/Utilities/NotifyIconDropHandler.cs:
```diff
@@ -0,0 +1,48 @@
+
+        public IDropInfo DropInFlight { get; set; }
+
+        public NotifyIconDropHandler(NotifyIconList notifyIconList)
+        {
+            _notifyIconList = notifyIconList;
+        }
```
?

dremin commented on RetroBar/Controls/NotifyIconList.xaml.cs:
```diff
@@ -174,6 +168,10 @@ private void NotificationArea_NotificationBalloonShown(object sender, Notificati
+
+            // Set up drag/drop handler
+            dropHandler = new NotifyIconDropHandler(this);
+            GongSolutions.Wpf.DragDrop.DragDrop.SetDropHandler(NotifyIcons, dropHandler);
```
Maybe this only needs to happen when dropHandler is null? The Loaded event fires whenever the theme changes, but we probably don't need to recreate this each time.

xoascf replied:
Thanks. 😅

dremin commented on RetroBar/Controls/NotifyIconList.xaml.cs:
```diff
@@ -230,5 +266,25 @@ private void SetToggleVisibility()
+
+        public void SaveIconOrder()
+        {
+            var visibleIcons = new System.Collections.Generic.List<ManagedShell.WindowsTray.NotifyIcon>();
```
It would be good to do this

xoascf replied:
Oh, okay, will check that!

xoascf replied:
Done!

xoascf committed:
Guard notify icon drop handler initialization

Wraps drag/drop setup in a null check so `NotifyIconDropHandler` is only created and assigned once during load. This prevents repeated handler reinitialization when the control is loaded multiple times.

xoascf committed:
Simplify NotifyIconList generic type usage

Add `using System.Collections.Generic` in `NotifyIconList.xaml.cs` and replace fully qualified `List<>`/`HashSet<>` references with the shorter type names. Way cleaner.

dremin reviewed (approved):
🎉 Thanks!
